### PR TITLE
Fix sync issues and shiftclick

### PR DESCRIPTION
**Pull request by falkreon**
* Syncs to all connected clients after a datapack live-reload
* Rework quickTransfer to allow shift-click into input slot.

### CODE REVIEW

File: src/main/java/io/github/debuggyteam/tablesaw/TableSawScreenHandler.java [hunk 1]
```diff
@@ -94,35 +98,78 @@ public ItemStack quickTransfer(PlayerEntity player, int fromIndex) {
...
+				// Item is coming from the inventory or hotbar, so try putting it into the input slot first.
+				
+				if (this.insertItem(result, INPUT_SLOT, INPUT_SLOT + 1, false)) {
+					// Success!
+					if (result.isEmpty()) {
+						//Very important to prevent dupes
+						slot.setStack(ItemStack.EMPTY);
+					}
+					
+					System.out.println("Count remaining: "+result.getCount()+" initial: "+initial.getCount());
```
Comment (woodiertexas) [inline src/main/java/io/github/debuggyteam/tablesaw/TableSawScreenHandler.java]: `System.out.println("Count remaining: "+result.getCount()+" initial: "+initial.getCount());` probably looks better as `System.out.println("Count remaining: " + result.getCount() + " initial: " + initial.getCount());`

Comment (falkreon) [reply src/main/java/io/github/debuggyteam/tablesaw/TableSawScreenHandler.java]: Sounds like I left a debug message in there by accident

Comment (woodiertexas) [reply src/main/java/io/github/debuggyteam/tablesaw/TableSawScreenHandler.java]: everyone's done that at least once lol

File: src/main/java/io/github/debuggyteam/tablesaw/TableSawRecipeSync.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+			PacketByteBuf buf = PacketByteBufs.create();
+			buf.writeVarInt(100);
+			for(int i = 0; i < 100; i++) {
+				writeRecipe(list.pop(), buf);
+			}
+
+			ServerPlayNetworking.send(player, TableSaw.RECIPE_CHANNEL, buf);
+		}
+
+		//cleanup any additional in a single packet
```
Comment (woodiertexas) [inline src/main/java/io/github/debuggyteam/tablesaw/TableSawRecipeSync.java]: forgot to specify what gets cleaned up lol

File: src/main/java/io/github/debuggyteam/tablesaw/TableSawScreenHandler.java [hunk 3]
```diff
@@ -94,35 +98,78 @@ public ItemStack quickTransfer(PlayerEntity player, int fromIndex) {
...
+			} else if (fromIndex == OUTPUT_SLOT) {
+				// Try to move the item from the output slot into the inventory, or the hotbar if no inventory slots are available.
+				
+				if (!this.insertItem(result, FIRST_INVENTORY_SLOT, LAST_HOTBAR_SLOT + 1, false)) {
 					return ItemStack.EMPTY;
 				}
-			} else if (fromIndex >= 2 && fromIndex < 29) {
-				if (!this.insertItem(result, 29, 38, false)) {
-					return ItemStack.EMPTY;
+			} else if (fromIndex >=FIRST_INVENTORY_SLOT && fromIndex <= LAST_HOTBAR_SLOT) {
```
Comment (woodiertexas) [inline src/main/java/io/github/debuggyteam/tablesaw/TableSawScreenHandler.java]: a formatting mistake (`fromIndex >=FIRST_INVENTORY_SLOT`)

**Review by woodiertexas — COMMENTED** [review]
a couple typos here and there

**Pull request merged**